### PR TITLE
Add busbar nodetypes command - Fetch data of available nodetypes on the server side

### PR DESCRIPTION
**Pull request by waldman**

### CODE REVIEW

File: lib/busbar_cli/repositories/nodetypes_repository.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  class << self
+    def all
+      nodetypes_data = JSON.parse(Request.get(NODETYPES_ROUTE).body)
+
+      nodetypes_data.map do |nodetype_data|
+        NodeType.new(nodetype_data)
+      end
+    end
+
+    # def find(nodetypes_id:) # For future implementation if needed
```
Comment (ilyakam) [inline lib/busbar_cli/repositories/nodetypes_repository.rb]: Is all of this commented-out code still necessary?

**Comment by waldman** [comment]
@markgergely @ilyakam I would like to leave it like that because to be able to fetch specific nodetype data it's a matter of just code it on the server side and then uncomment on the CLI. (so... IMHO I can't see why not leave it there).

But up to you guys... anything I can just remove.

File: lib/busbar_cli.rb [hunk 2]
```diff
@@ -46,4 +46,5 @@ class BusbarCLI < Thor
   include Commands::BusbarVersion
   include Commands::Copy
   include Commands::Exec
+  include Commands::NodeTypes
```
Comment (ilyakam) [inline lib/busbar_cli.rb]: Please choose between one word (`nodetypes`) or two words (`NodeTypes`) and maintain consistency throughout this PR. It looks like the command/configuration setting is `nodetypes`, in which case it should read `Commands:Nodetypes` on this line. This comment applies to the entire PR.

File: lib/busbar_cli/models/nodetype.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class NodeType
+  include Virtus.model
+
+  attribute :id, String
+  attribute :cpu, String
+  attribute :guaranteed_cpu, String
+  attribute :memory, String
+
+  def as_text
+    "#{id} =>\tCPU: #{cpu} \tMemory: #{memory}"
```
Comment (ilyakam) [inline lib/busbar_cli/models/nodetype.rb]: Must there be a space before the tab?
```suggestion
    "#{id} =>\tCPU: #{cpu}\tMemory: #{memory}"
```

**Comment by markgergely** [comment]
up to you, i dont have a strong opinion here :)

**Comment by waldman** [comment]
In case we need to fetch single nodetypes (and code the needed support for the endpoint on the server side) we can use the following code snippet on the `lib/busbar_cli/repositories/nodetypes_repository.rb` file:
```
    def find(nodetypes_id:) # For future implementation if needed
      @nodetype_id = nodetype_id

      request = Request.get(nodetypes_route)

      return if request.code == '404'

      nodetype.new(JSON.parse(request.body)['data'])
    end

    private

    def nodetypes_route
      "#{NODETYPES_ROUTE}#{@nodetype_id}"
    end
```

**Pull request merged**